### PR TITLE
fix: [CDS-83627]: fixing p3 bugs around snow update/create

### PR DESCRIPTION
**Pull request by satyam-harness**
### Summary
Bugs - 

- ReadOnly fields call should only be made when standard tab is selected, Not be called when configure fields or form template is selected. (https://harness.atlassian.net/browse/CDS-83633)
- ServiceNow create using Standard template is blocked on ui when ticket type is change_request (Capital Sensitive). (https://harness.atlassian.net/browse/CDS-83631)
- Template field in standard tab in FE should be required. (https://harness.atlassian.net/browse/CDS-83630)

Solution - 

- Readonly call is now made when we switch to standard template checkbox.
- before matching ticketType value with `change_request` key, we are calling `ticket_type.toLowerCase()` then matching.
- Added validation on template fields for both FF off (condition on useTemplate) and FF on (condition on createType).

Test cases - 

- Added testcase to check validation of template field.

Manual Testing - 

- readonly call is not called when configure field or form template is seclected.
- readonly call is made upon initial loading of step if standard change template was selected.
- template field validation is not run when we switch to configure fields.
- tested working of standard change template for ticket_types - `Change_request`, `CHANGE_REQUEST`, `change_request`, `Change_Request`.

#### Screenshots

https://github.com/harness/harness-core-ui/assets/96036463/64ef2963-4d77-469f-8378-36ee6327016b

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger eslint

**Comment by satyam-harness** [comment]
trigger messagecheck

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx [hunk 1]
```diff
@@ -192,9 +193,15 @@ function FormContent({
     return !fetchingTemplate && !fetchTemplateError && isEmpty(templateResponse)
   }, [fetchingTemplate, fetchTemplateError, templateResponse])
 
+  const isTcketTypeChangeRequest =
+    (defaultTo(get(formik, 'values.spec.ticketType'), '') as string).toLowerCase() === 'change_request'
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:197]: No need to use default to in this scenario - third argument of lodash get can take care of default value
get(formik, 'values.spec.ticketType', '')

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx [hunk 2]
```diff
@@ -192,9 +193,15 @@ function FormContent({
...
+
   useEffect(() => {
     if (connectorRefFixedValue && connectorValueType === MultiTypeInputType.FIXED) {
-      if (CDS_GET_SERVICENOW_STANDARD_TEMPLATE) {
+      if (
+        CDS_GET_SERVICENOW_STANDARD_TEMPLATE &&
+        formik.values.spec.fieldType === FieldType.CreateFromStandardTemplate
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:203]: can spec be undefined here ? please check for NPE

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowFieldsRenderer.tsx [hunk 3]
```diff
@@ -81,15 +84,19 @@ function GetMappedFieldComponent({ selectedField, props, expressions, index }: M
         name={`spec.selectedFields[${index}].value`}
         placeholder={selectedField.name}
         multiTypeTextArea={{ enableConfigureOptions: false, expressions, allowableTypes: props.allowableTypes }}
-        disabled={isApprovalStepFieldDisabled(props.readonly || selectedField.readOnly)}
+        disabled={isApprovalStepFieldDisabled(
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowFieldsRenderer.tsx:87]: Lets extract it out in a variable and use it at all places

isApprovalStepFieldDisabled(
          props.readonly || (props.serviceNowType === SERVICENOW_TYPE.UPDATE && selectedField.readOnly)
        )

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx]: okay, done

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx]: added for safety.

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowFieldsRenderer.tsx (hunk 3 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowFieldsRenderer.tsx]: yes, makes sense. thanks

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx [hunk 4]
```diff
@@ -192,9 +193,14 @@ function FormContent({
     return !fetchingTemplate && !fetchTemplateError && isEmpty(templateResponse)
   }, [fetchingTemplate, fetchTemplateError, templateResponse])
 
+  const isTcketTypeChangeRequest = (get(formik, 'values.spec.ticketType') as string).toLowerCase() === 'change_request'
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:196]: spelling error -> isTicketTypeChangeRequest

**Review by samarth-sharma31 — APPROVED** [review]

**Comment by satyam-harness** [comment]
retrigger prettier

**Comment by satyam-harness** [comment]
retrigger cypress-cv

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx (hunk 4 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx]: 🥲thanks

**Review by Eric761 — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Pull request merged**